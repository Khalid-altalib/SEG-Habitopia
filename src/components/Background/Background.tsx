import React from "react";
import { LinearGradient } from "expo-linear-gradient";
import { theme, useColorModeValue } from "native-base";

/**
 * Displays a background, that is either dark or light depending on the color scheme, with a slight
 * gradient.
 *
 * @returns The background component.
 */
const Background = () => {
  const LIGHT_MODE_GRADIENT = [
    theme.colors.blueGray[200],
    theme.colors.blueGray[100],
  ];
  const DARK_MODE_GRADIENT = [
    theme.colors.blueGray[900],
    theme.colors.blueGray[800],
  ];

  return (
    <LinearGradient
      colors={useColorModeValue(LIGHT_MODE_GRADIENT, DARK_MODE_GRADIENT)}
      style={{
        flex: 1, // Make the background fill all of the space available to it
      }}
    />
  );
};

export default Background;
