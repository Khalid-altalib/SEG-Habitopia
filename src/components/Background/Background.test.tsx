// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Habitopia
import Background from "@components/Background";

it("Renders correctly.", () => {
  render(<Background>This is some example content.</Background>);

  expect(screen.toJSON()).toMatchSnapshot();
});
