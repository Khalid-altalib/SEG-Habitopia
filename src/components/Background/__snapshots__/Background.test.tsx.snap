// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4293190887,
        4294243573,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    {
      "height": "100%",
    }
  }
>
  This is some example content.
</ViewManagerAdapter_ExpoLinearGradient>
`;
